lib/src/client/dio_interceptor.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:math';
+
+import 'package:cryptography/dart.dart';
+import 'package:dio/dio.dart';
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+
+/// Interceptor for signing HTTP requests with HMAC authentication.
+///
+/// This interceptor adds HMAC authentication headers to outgoing HTTP requests
+/// based on the provided client ID and client secret. It calculates the signature
+/// using the specified hash algorithm.
+class HMACDioInterceptor extends Interceptor {
```

Comment:
You're missing the `RequestsSignatureOptions` that the .NET implementation has to customize the behavior of the interceptor.
Because of that, you're missing some features such as the `ClockSkew`, `SignaturePattern`, etc.

Again, I don't understand why you didn't replicate the .NET implementation.

Reply:
I though this PR was just about the Dio interceptor?

Reply:
The Dio interceptor is the homologue of the .NET `RequestsSignatureDelegatingHandler` and they need to have the same features.

Reply:
My bad! Should have double checked.